server/src/main/java/com/cloud/configuration/ConfigurationManagerImpl.java

```diff
@@ -5032,8 +5032,12 @@ public NetworkOffering createNetworkOffering(final CreateNetworkOfferingCmd cmd)
             forVpc = false;
         }
 
-        final NetworkOffering offering = createNetworkOffering(name, displayText, trafficType, tags, specifyVlan, availability, networkRate, serviceProviderMap, false, guestType, false,
+        final NetworkOfferingVO offering = createNetworkOffering(name, displayText, trafficType, tags, specifyVlan, availability, networkRate, serviceProviderMap, false, guestType, false,
                 serviceOfferingId, conserveMode, serviceCapabilityMap, specifyIpRanges, isPersistent, details, egressDefaultPolicy, maxconn, enableKeepAlive, forVpc, domainIds, zoneIds);
+        if (enable) {
```

Comment:
Instead of updating the offeringVO again, pass the value of "enable" to createNetworkOffering() method and use offeringFinal.setState() method to set it while creating the initial VO entry itself.